src/Pickles/Pickles/Runner.cs

```diff
@@ -70,9 +70,8 @@ private static void ApplyTestResultsToFeatures(IContainer container, IConfigurat
         {
             var testResults = container.Resolve<ITestResults>();
 
-            foreach (var node in features)
+            foreach (var featureTreeNode in features.Where(p => p.GetType() == typeof(FeatureNode)).Select(p => (FeatureNode)p))
             {
-                var featureTreeNode = node as FeatureNode;
```

Comment:
This change has nothing to do with the fix, has it? Please do me a favour and resist giving in to ReSharper's style suggestions while doing a fix. I will be happy to review this kind of changes but please send them in a separate pull request.